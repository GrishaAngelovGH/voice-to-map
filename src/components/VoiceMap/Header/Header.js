import { useState } from 'react'
import Fab from '@mui/material/Fab'
import Box from '@mui/material/Box'

const Header = () => {
    const [isRecoding, setIsRecording] = useState(false)

    return (
        <Box
            sx={{
                height: 150,
                backgroundColor: 'primary.dark'
            }}
            className="row g-0 justify-content-center align-items-center p-3"
            data-testid="header"
        >
            <div className="col-md-7 text-white">
                <h1>Voice-To-Map</h1>
                <h3>Navigate map with your voice</h3>
            </div>
            <div className="col-md-5">
                <Fab variant="extended" color={!isRecoding ? "inherit" : "warning"}>
                    {!isRecoding ? "Record" : "Recording..."}
                </Fab>
            </div>
        </Box>
    )
}

export default Header