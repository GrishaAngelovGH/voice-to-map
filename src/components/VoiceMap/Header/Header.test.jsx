import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import Header from './Header'

test('should render Header component', () => {
	const mockStore = configureStore()
	const store = mockStore({ locations: { lastLocation: '', history: [] } })

	const view = render(
		<Provider store={store}>
			<Header />
		</Provider>
	)

	expect(view).toMatchSnapshot()
})
