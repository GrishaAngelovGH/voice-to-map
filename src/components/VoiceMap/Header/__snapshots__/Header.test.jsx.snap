// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Header component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-mq2wc9"
      >
        <div
          class="row g-0 justify-content-center align-items-center text-center p-3"
        >
          <div
            class="col-md-4 text-white border rounded bg-primary"
          >
            <h1
              class="m-0"
            >
              Voice-To-Map
            </h1>
            <h3
              class="m-0"
            >
              Navigate map with your voice
            </h3>
          </div>
          <div
            class="col-md-4"
          >
            <button
              class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-colorInherit MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-colorInherit mt-1 css-5w0vct-MuiButtonBase-root-MuiFab-root"
              tabindex="0"
              type="button"
            >
              Record Voice
            </button>
          </div>
          <div
            class="col-md-4 bg-light p-3 rounded"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorInfo Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-ipdm92-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Detected Input
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorInfo Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-18j0pu3-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id=":r0:"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-81qg8w"
                  >
                    <span>
                      Detected Input
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-mq2wc9"
    >
      <div
        class="row g-0 justify-content-center align-items-center text-center p-3"
      >
        <div
          class="col-md-4 text-white border rounded bg-primary"
        >
          <h1
            class="m-0"
          >
            Voice-To-Map
          </h1>
          <h3
            class="m-0"
          >
            Navigate map with your voice
          </h3>
        </div>
        <div
          class="col-md-4"
        >
          <button
            class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-colorInherit MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-colorInherit mt-1 css-5w0vct-MuiButtonBase-root-MuiFab-root"
            tabindex="0"
            type="button"
          >
            Record Voice
          </button>
        </div>
        <div
          class="col-md-4 bg-light p-3 rounded"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorInfo Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-ipdm92-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Detected Input
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorInfo Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-18j0pu3-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-81qg8w"
                >
                  <span>
                    Detected Input
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
